sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5664,6 +5664,18 @@ pi_result piextEventCreateWithNativeHandle(pi_native_handle NativeHandle,
   // need to create a host-visible proxy for it.
   (*Event)->HostVisibleEvent = *Event;
 
+  // Regular events managed by SYCL RT are created with RefCnt == 2 to represent
+  // that their resource is "owned" by the queue where the task that would
+  // signal this event is running. There is no such implicit ownership with the
+  // events created from native handle. If we don't own it then it's completely
+  // user's responsibility and if we do then we have no way to determine if such
+  // a task exists. Indeed, the user might have created ZeEvent in non-signaled
+  // state and never used it to run anything. As such, even in the scenario with
+  // ownership transfer, it's user's responsibility to ensure that lifetime of
+  // the sycl::even they are creating covers the point where anything signals
+  // it.
```

Comment:
I agree that we should not expect to see a "signal" of this event, or "wait" on it without user's explicit ask for that (e.g. using this event as a dependency of another event). But we don't also require the user to signal event. Instead, disregarding of whether it was signaled or not, we should delete PI event when SYCL event goes out of its scope (but only destroy L0 native event if the ownership was transferred to SYCL RT). I think at least the comment needs to be refined.